apps/files/js/breadcrumb.js

```diff
@@ -163,19 +189,20 @@
 				if(menuPart.dir) {
 					$menuItem = $('<li class="crumblist"><a><span class="icon-folder"></span><span></span></a></li>');
 					$menuItem.data('dir', menuPart.dir);
-					$menuItem.find('a').attr('href', this.getCrumbUrl(part, j));
+					$menuItem.find('a').attr('href', this.getCrumbUrl(parts[Math.max(this.activeItemIndex, 0)], j));
```

Comment:
@skjnldsv 

You introduce `this.getCrumbUrl(part, j));` in https://github.com/nextcloud/server/pull/7051. Did you mean `menuPart` instead of `part`?

`part` would mean the latest item in the previous for loop. `menuPart` would mean the current item in the current loop.

If `menuPart`, then let's use `menuPart` instead of the proposed change.

Reply:
This part felt very odd to be honest. I thought that this is because of the "old fashioned style" in this code - but I agree, `menuPart` feels more natural here.

Reply:
> You introduce this.getCrumbUrl(part, j)); in https://github.com/nextcloud/server/pull/7051. Did you mean menuPart instead of part?

It was in 2017, safe to say I don't recall at all :see_no_evil:

Reply:
Just checked on master, and when the breadcrumb is squeezed into '...', items that are in the '...' all link to the same URL. So it feels like a never noticed bug.

Let's use `menuPart` @paierlep :).

```suggestion
 					$menuItem.find('a').attr('href', this.getCrumbUrl(menuPart, j)); 
```

Reply:
Yes - I've already changed it as my tests also showed that `menuPart` should be correct!